amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/DiagnosticEventsTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.kinesis.coordinator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import software.amazon.kinesis.leases.Lease;
+import software.amazon.kinesis.leases.LeaseBuilder;
+import software.amazon.kinesis.leases.LeaseCoordinator;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+@RunWith(MockitoJUnitRunner.class)
+public class DiagnosticEventsTest {
+    @Mock
+    private ThreadPoolExecutor executor;
+    @Mock
+    private LeaseCoordinator leaseCoordinator;
+    @Mock
+    private DiagnosticEventHandler defaultHandler;
+
+    private DiagnosticEventHandler customHandler = new CustomHandler();
+    private boolean wasCustomHandlerInvoked;
+
+    private final Throwable throwable = new TestRejectedTaskException();
+
+    private final int activeThreadCount = 2;
+    private final int corePoolSize = 4;
+    private final int largestPoolSize = 8;
+    private final int maximumPoolSize = 16;
+
+    private SynchronousQueue<Runnable> executorQueue;
+    private Collection<Lease> leaseAssignments;
+
+    @Before
+    public void setup() {
+        wasCustomHandlerInvoked = false;
+
+        executorQueue = new SynchronousQueue<>();
+
+        final Lease lease = new LeaseBuilder().build();
+        leaseAssignments = Collections.singletonList(lease);
+
+        when(executor.getQueue()).thenReturn(executorQueue);
+        when(executor.getActiveCount()).thenReturn(activeThreadCount);
+        when(executor.getCorePoolSize()).thenReturn(corePoolSize);
+        when(executor.getLargestPoolSize()).thenReturn(largestPoolSize);
+        when(executor.getMaximumPoolSize()).thenReturn(maximumPoolSize);
+        when(leaseCoordinator.getAssignments()).thenReturn(leaseAssignments);
+    }
+
+    @Test
+    public void testExecutorStateEventWithDefaultHandler() {
```

Comment:
Avoid the term default. `testExecutorStateEvent()` should be fine as you are passing in a handler that is not a custom handler.

Reply:
Renamed without 'default'